bin/middlewares/fullText_query.js

```diff
@@ -0,0 +1,71 @@
+const axios = require('axios');
+const yaml = require('node-yaml');
+
+module.exports = function (config) {
+  const idunnBaseUrl = config.server.services.idunn.url || config.services.idunn.url;
+  const geocoderUrl = idunnBaseUrl + '/v1/autocomplete';
+  const useNlu = config.services.geocoder.useNlu;
+  const idunnTimeout = Number(config.server.services.idunn.timeout);
+
+  const categories = yaml.readSync('../../config/categories.yml');
+  const isCategoryValid = type => categories.find(category => category.name === type);
+
+  const removeNullEntries = obj =>
+    Object.entries(obj)
+      .filter(([_key, value]) => value !== null && value !== undefined)
+      .reduce((result, [key, value]) => ({ ...result, [key]: value }), {});
+
+  if (isNaN(idunnTimeout)) {
```

Comment:
we might move this block right after l.8 ? In order to raise error earlier, and handle the NaN case next to the Number construction

Reply:
Yep why not.